### ISSUE TITLE
[Bug]: Shading and unshading windows corrupts the tiling

### ISSUE THREAD

**Issue by ilkerhk** [gikari/bismuth#65]
### Summary

Shading a window is not working as expected.

### Steps to Reproduce

1. Start tiling with at least 2 windows
2. shade a window
3. When I shade a window (for example one located in the lower part of the screen), the remaining tiled windows do not resize. I can see the desktop under the previously shaded window.

### Expected behavior

When a window is shaded the area under it should be used by the other windows.
Best solution (for me as a user) is that after the shading, the whole tiling and sizes of the windows should be updated (now shaded window has a little height, and should be tiled with this low height).
Or alternative solutions is that: shaded window should switch to floating mode (kwin-tiling script does that), and unshading should put it back into tiling.

### Bismuth version

NA

### KDE Plasma version

5.22

### The platform KWin is running on

X11

**Comment by gikari** [comment]
In the dynamic adjust scenario, what should happen in the situation like this (The window on the left is shaded):

![Screenshot_20210928_153035](https://user-images.githubusercontent.com/14205339/135086963-d794d3cc-aea7-44ff-b5ae-8d57359e0645.png)

The left side is a "master area" and technically the windows on the left do not belong to it. Should windows on the right move to the master area?

**Comment by ilkerhk** [comment]
Yes, thats a good point. In my opinion, yes, one of the right windows should move to the left. 
If a user is using 3 windows and decides to shade master, then I think that he/she probably wants to use one of the others in a larger window temporarily.

I usually use shading when I have several windows but only one needed at a time (such as several graphs or images). Finding a window and unshading when needed is much faster compared to minimize/unminimize.

**Comment by ilkerhk** [comment]
Thanks for the reply. 
How about the following solution: 
If a window is shaded (while it is tilled), simply switching the window to floating mode.   
When a window is unshaded then it can switch back to its original mode before shading. 
Would that be possible?

**Comment by gikari** [comment]
It is possible, if it fixes the above bug, I would like to implement it.

**Comment by ilkerhk** [comment]
Yes, I think that would fix the bug (it is one of the requested solutions in the OP).  
When a window is shaded it will switch to floating mode and therefore the rest of the windows will be rearranged so that they tile the whole screen.
